mesa/agent.py

```diff
@@ -179,7 +179,7 @@ def agent_generator(filter_func=None, agent_type=None, n=0):
 
         return AgentSet(agents, self.model) if not inplace else self._update(agents)
 
-    def shuffle(self, inplace: bool = False) -> AgentSet:
+    def shuffle(self, inplace: bool = True) -> AgentSet:
```

Comment:
This is an API change - not necessarily against it, but we have to go thought a proper announce and release process. Do you think that's worth it? Also, is it consistent with other `inplace` default values in the AgentSet methods?

Reply:
Hence my comment. I haven't checked the rest.

The issue is that for e.g., select defaulting to new makes sense. While for shuffle inplace might make more sense. 

One solution is to go back to the old behavior and just document the performance benefit of inplace. Let's not overthink this.

Reply:
I would go for that approach. `select()` and `sort()` currently also default to `inplace=False`, and if I recall correctly (most of) Pandas does that to. The consistency there is worth something.

But yes, let's document in the docstring that inplace shuffling is faster and demonstrate that also in our public examples (if we have any already).

Reply:
I'll do that later today.

Reply:
Do I change https://github.com/projectmesa/mesa-examples/blob/479eaf389e1a98bda0bdba5aa8f24ede0d84e764/examples/schelling_experimental/model.py#L70 to `self.agents.shuffle(inplace=True).do("step")`?

Reply:
Yes that should save some time. I haven't fully benchmarked it but inplace avoids unpacking and repacking of weakrefs.